src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/RazorUpdaterStep.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Razor.Extensions;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Razor
+{
+    /// <summary>
+    /// Upgrade step that updates cshtml source using sub-steps that
+    /// analyze RazorCodeDocument objects and make necessary updates
+    /// to cshtml files.
+    /// </summary>
+    public class RazorUpdaterStep : UpgradeStep
+    {
+        private RazorProjectEngine? _razorEngine;
+
+        public override string Title => "Update Razor files";
+
+        public override string Description => "Update Razor files using registered Razor updaters";
+
+        public override string Id => WellKnownStepIds.RazorUpdaterStepId;
+
+        public override IEnumerable<string> DependsOn { get; } = new[]
+        {
+            // Project should be backed up before changing source
+            WellKnownStepIds.BackupStepId,
+
+            // Template files should be added prior to changing source (since some code fixers will change added templates)
+            WellKnownStepIds.TemplateInserterStepId,
+
+            // Project should have correct TFM
+            WellKnownStepIds.SetTFMStepId,
```

Comment:
We've been including all these, but are they necessary? What does this step actually require to complete?

Reply:
I've tried to include the ones that we actually do want run first but it's certainly possible this list isn't right. My reasoning for these was:

1. Backup should run first since we'll be changing files.
2. Template inserter should run first since some of the source updaters depend on there being a startup.cs they can modify.
3. Set TFM should run first because most Razor updaters (applying existing code fixes to cshtml and replacing @Helper sections) will be updating be add .NET Core-specific code. Eventually we may want Razor updaters that can run before and after TFM change. That seems a very low priority at this point, though.

Reply:
Hmm I see that reasoning. It just feels like it's extra, but that makes sense.